### PR TITLE
stripping comments from JSON files upon execution

### PR DESCRIPTION
**Pull request by ds-ms**
I am using an external library. This version of it has been scanned. Here's a [link](https://ossmsft.visualstudio.com/_oss?searchText=strip-json&_a=componentversion&cvid=795&fullPageArtifact=true) to it.

### CODE REVIEW

File: Tasks/AzureResourceGroupDeployment/package.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 
 {
-  "name": "DeployAzureResourceGroup",
+  "name": "AzureResourceGroupDeployment",
   "main":"main.js",
   "dependencies": {
     "vsts-task-lib" : "^0.9.20",
-    "vso-node-api": "6.0.1-preview"
+    "vso-node-api": "6.0.1-preview",
+    "strip-json-comments": "1.0.4"
```
Comment (kmkumaran) [inline Tasks/AzureResourceGroupDeployment/package.json]: Do we know what are the nested dependencies this brings? Can you double check the OSS requirements like if the version you are referring is HIBS scanned and if it is tracked in our OSS tool.

Comment (kmkumaran) [inline Tasks/AzureResourceGroupDeployment/package.json]: You would also want to update third party notice accordingly.

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 2]
```diff
@@ -219,7 +220,7 @@ export class ResourceGroup {
         var template: Object;
         try {
             tl.debug("Loading CSM Template File.. " + this.taskParameters.csmFile);
-            template = JSON.parse(fileEncoding.readFileContentsAsText(this.taskParameters.csmFile));
+            template = JSON.parse(stripJsonComments(fileEncoding.readFileContentsAsText(this.taskParameters.csmFile)));
```
Comment (kmkumaran) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: Do you have any L0 tests? if there are existing tests, you can update them to have comments.

File: Tasks/AzureResourceGroupDeployment/package.json (hunk 1 above)
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/package.json]: This doesn't bring any dependencies. And yes, it's HIBS scanned. I have specifically used the one which was scanned

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts (hunk 2 above)
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: Sure

**Pull request merged**